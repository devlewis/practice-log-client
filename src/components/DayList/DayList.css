.container {
  display: flex;
  justify-content: space-around;
  flex-wrap: wrap;
  align-content: flex-start;
  margin-bottom: 30px;
  background-color: lightgray;
}

.daylisth1 {
  margin-top: 0;
}

.container a {
  text-decoration: none;
  color: black;
}

.logout_btn {
  margin-top: 10px;
  margin-bottom: 15px;
}

.viewbox {
  display: flex;
  justify-content: center;
}

.viewbtn {
  margin-bottom: 32px;
}

.navigators {
  display: flex;
  justify-content: space-evenly;
  margin-bottom: 10px;
}

.toprow {
  display: flex;
  flex-direction: column;
}

.tothours {
  background-color: aliceblue;
  border-radius: 10%;
  font-weight: 700;
  font-size: 1.25em;
  align-self: center;
  padding: 5px 10px;
  margin:15px 0;
}

h1 {
  text-align: center;
  margin-bottom: 0;
}

@media screen and (min-width: 415px) {

  .toprow {
    margin-bottom: 0px;
  }

  .fa-2x:hover {
    color: gray;
    cursor: pointer;
  }
}
