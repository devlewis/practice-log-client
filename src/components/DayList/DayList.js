import React, { PureComponent } from "react";
import Day from "../Day/Day";
import "./DayList.css";
import FinalView from "../FinalView/FinalView";
import TokenService from "../../services/token-service";
import { Link } from "react-router-dom";
import Context from "../../Context";

class DayList extends PureComponent {
  //state= { page: 0 }

  static contextType = Context;

  handleLogoutClick = () => {
    TokenService.clearAuthToken();
  };

  render() {
    const days = this.context.days.map((day) => (
      <Day key={day.id} {...day} goal_hours={this.context.hours_goal} />
    ));

    return (
      <div>
        <div>
          Total Hours:
          {this.context.total_hours}
        </div>
        <div className="container">{days}</div>
        {this.context.days[this.context.days.length - 1].completed === true && (
          <FinalView
            totalDays={this.props.num_of_days}
            onClickSetup={({ history }) => history.push("/setup")}
          />
        )}
        {TokenService.hasAuthToken() && (
          <Link to="/">
            <button onClick={this.handleLogoutClick}>Logout</button>
          </Link>
        )}
      </div>
    );
  }
}
export default DayList;
