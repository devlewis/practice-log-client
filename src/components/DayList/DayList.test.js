import React from "react";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";
import DayList from "./DayList";
import Context from "../../Context";
import { Route, BrowserRouter } from "react-router-dom";

let value = {
  days: [
    {
      id: 1,
      day_num: 1,
      date: "March 22",
      completed: "false",
      technique: "",
      repertoire: "",
      actual_hours: 0,
      touched: false,
      goal_id: 1,
      user_id: 1,
      goal_hours: 3,
    },
    {
      id: 2,
      day_num: 2,
      date: "March 23",
      completed: "true",
      technique: "Hanon",
      repertoire: "",
      actual_hours: 2,
      touched: true,
      goal_id: 1,
      user_id: 1,
      goal_hours: 3,
    },
  ],
  total_hours: 5,
  hours_goal: 3,
  num_of_days: 2,
};

describe(`DayList component`, () => {
  const props = {
    className: "test-class-name",
    children: <p>test children</p>,
    "data-other": "test-other-prop",
  };

  it("renders a list DayList by default", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Route>
          <Context.Provider value={value}>
            <DayList />
          </Context.Provider>
        </Route>{" "}
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("renders a list DayList given props", () => {
    const wrapper = mount(
      <BrowserRouter>
        <Route>
          <Context.Provider value={value}>
            <DayList {...props} />
          </Context.Provider>
        </Route>
      </BrowserRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
