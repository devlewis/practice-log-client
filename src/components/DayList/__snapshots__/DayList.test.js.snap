// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DayList component renders a list DayList by default 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route>
      <DayList>
        <div>
          <FinalView
            onClickSetup={[Function]}
            totalDays={2}
          >
            <main>
              <h1>
                YOU DID IT!
              </h1>
              <p
                className="finalview_p"
              >
                You practiced 
                1
                 out of 
                2
                 
                goal days!
              </p>
              <p
                className="finalview_p"
              >
                You reached your practice hourly goal 
                0
                 out of 
                2
                 days.
              </p>
              <p
                className="finalview_p"
              >
                 
                You practiced a total of 
                5
                 hours during your
                 
                2
                -day goal.
              </p>
              <p
                className="finalview_p"
              >
                Technique items recorded:
              </p>
              <ul
                className="technique_list"
              >
                <li
                  key="0"
                >
                  Hanon
                </li>
              </ul>
              <p
                className="finalview_p"
              >
                Repertoire items recorded:
              </p>
              <ul
                className="repertoire_list"
              />
              <Link
                to="/setup"
              >
                <LinkAnchor
                  href="/setup"
                  navigate={[Function]}
                >
                  <a
                    href="/setup"
                    onClick={[Function]}
                  >
                    <button>
                      Set up a new goal!
                    </button>
                  </a>
                </LinkAnchor>
              </Link>
            </main>
          </FinalView>
          <div>
            Total Hours:
            5
          </div>
          <button
            onClick={[Function]}
          >
            Start over with a new goal!
          </button>
          <div
            className="viewbox"
          />
          <div
            className="navigators"
          />
          <div
            className="container"
          >
            <Day
              actual_hours={0}
              completed="false"
              date="March 22"
              day_num={1}
              goal_hours={3}
              goal_id={1}
              hours_goal={3}
              id={1}
              key="1"
              repertoire=""
              technique=""
              touched={false}
              user_id={1}
            >
              <div
                className="day_box"
                style={Object {}}
              >
                <p>
                  Day #
                  1
                </p>
                <p />
                <Link
                  to="/day/1"
                >
                  <LinkAnchor
                    href="/day/1"
                    navigate={[Function]}
                  >
                    <a
                      href="/day/1"
                      onClick={[Function]}
                    >
                      Did you practice today?
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Day>
            <Day
              actual_hours={2}
              completed="true"
              date="March 23"
              day_num={2}
              goal_hours={3}
              goal_id={1}
              hours_goal={3}
              id={2}
              key="2"
              repertoire=""
              technique="Hanon"
              touched={true}
              user_id={1}
            >
              <div
                className="day_box"
                style={
                  Object {
                    "backgroundColor": "yellow",
                    "color": "black",
                  }
                }
              >
                <p>
                  Day #
                  2
                </p>
                <p />
                <section
                  className="completed_box"
                >
                  <p>
                    Hours Practiced: 
                    2
                  </p>
                  <p>
                    Goal Hours: 
                    3
                  </p>
                  <p>
                    Practice Details:
                  </p>
                  <p>
                    Technique: 
                    Hanon
                  </p>
                  <Link
                    className="small"
                    to="/day/2"
                  >
                    <LinkAnchor
                      className="small"
                      href="/day/2"
                      navigate={[Function]}
                    >
                      <a
                        className="small"
                        href="/day/2"
                        onClick={[Function]}
                      >
                        Edit Practice Details
                      </a>
                    </LinkAnchor>
                  </Link>
                </section>
              </div>
            </Day>
          </div>
        </div>
      </DayList>
    </Route>
     
  </Router>
</BrowserRouter>
`;

exports[`DayList component renders a list DayList given props 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Route>
      <DayList
        className="test-class-name"
        data-other="test-other-prop"
      >
        <div>
          <FinalView
            onClickSetup={[Function]}
            totalDays={2}
          >
            <main>
              <h1>
                YOU DID IT!
              </h1>
              <p
                className="finalview_p"
              >
                You practiced 
                1
                 out of 
                2
                 
                goal days!
              </p>
              <p
                className="finalview_p"
              >
                You reached your practice hourly goal 
                0
                 out of 
                2
                 days.
              </p>
              <p
                className="finalview_p"
              >
                 
                You practiced a total of 
                5
                 hours during your
                 
                2
                -day goal.
              </p>
              <p
                className="finalview_p"
              >
                Technique items recorded:
              </p>
              <ul
                className="technique_list"
              >
                <li
                  key="0"
                >
                  Hanon
                </li>
              </ul>
              <p
                className="finalview_p"
              >
                Repertoire items recorded:
              </p>
              <ul
                className="repertoire_list"
              />
              <Link
                to="/setup"
              >
                <LinkAnchor
                  href="/setup"
                  navigate={[Function]}
                >
                  <a
                    href="/setup"
                    onClick={[Function]}
                  >
                    <button>
                      Set up a new goal!
                    </button>
                  </a>
                </LinkAnchor>
              </Link>
            </main>
          </FinalView>
          <div>
            Total Hours:
            5
          </div>
          <button
            onClick={[Function]}
          >
            Start over with a new goal!
          </button>
          <div
            className="viewbox"
          />
          <div
            className="navigators"
          />
          <div
            className="container"
          >
            <Day
              actual_hours={0}
              completed="false"
              date="March 22"
              day_num={1}
              goal_hours={3}
              goal_id={1}
              hours_goal={3}
              id={1}
              key="1"
              repertoire=""
              technique=""
              touched={false}
              user_id={1}
            >
              <div
                className="day_box"
                style={Object {}}
              >
                <p>
                  Day #
                  1
                </p>
                <p />
                <Link
                  to="/day/1"
                >
                  <LinkAnchor
                    href="/day/1"
                    navigate={[Function]}
                  >
                    <a
                      href="/day/1"
                      onClick={[Function]}
                    >
                      Did you practice today?
                    </a>
                  </LinkAnchor>
                </Link>
              </div>
            </Day>
            <Day
              actual_hours={2}
              completed="true"
              date="March 23"
              day_num={2}
              goal_hours={3}
              goal_id={1}
              hours_goal={3}
              id={2}
              key="2"
              repertoire=""
              technique="Hanon"
              touched={true}
              user_id={1}
            >
              <div
                className="day_box"
                style={
                  Object {
                    "backgroundColor": "yellow",
                    "color": "black",
                  }
                }
              >
                <p>
                  Day #
                  2
                </p>
                <p />
                <section
                  className="completed_box"
                >
                  <p>
                    Hours Practiced: 
                    2
                  </p>
                  <p>
                    Goal Hours: 
                    3
                  </p>
                  <p>
                    Practice Details:
                  </p>
                  <p>
                    Technique: 
                    Hanon
                  </p>
                  <Link
                    className="small"
                    to="/day/2"
                  >
                    <LinkAnchor
                      className="small"
                      href="/day/2"
                      navigate={[Function]}
                    >
                      <a
                        className="small"
                        href="/day/2"
                        onClick={[Function]}
                      >
                        Edit Practice Details
                      </a>
                    </LinkAnchor>
                  </Link>
                </section>
              </div>
            </Day>
          </div>
        </div>
      </DayList>
    </Route>
  </Router>
</BrowserRouter>
`;
